test/libsolidity/syntaxTests/types/magic_block_blobbasefee_error.sol

```diff
@@ -0,0 +1,16 @@
+contract C {
+    function f() public view returns (uint) {
+        return block.blobbasefee;
+    }
+    function g() public view returns (uint ret) {
+        assembly {
+            ret := blobbasefee()
+        }
+    }
+}
+// ====
+// EVMVersion: <=shanghai
+// ----
+// TypeError 1006: (74-91): "blobbasefee" is not supported by the VM version.
+// DeclarationError 4619: (187-198): Function "blobbasefee" not found.
```

Comment:
Something must be wrong here. I'd expect this test to trigger error 6679 (`only available for Cancun-compatible VMs`), but it instead reports 4619 (`Function ... not found`). `blobbasefee` should be recognized as a built-in even if it's rejected, just as it is for `basefee`.

Note that this not happen for `basefee`. If you replace `blobbasefee` with `basefee` and run this test with `--evm-version berlin`, it generates error 5430 (`only available for London-compatible VMs`) as expected.

This discrepancy on its own is not a big issue, but if it's a bug, something else could be broken so I think it's better to check why this happens.

And by the way, since 5430 seems reproducible, we could add back the same case for it too and remove it from `error-codes.py`. It was removed in #11763, but I'm not sure why. This makes me wonder if this isn't something weird on my side, but I rechecked multiple times, even with a release binary, and I can trigger this error every time.

Reply:
For `basefee` actually, the assembly block was explicitly removed - https://github.com/ethereum/solidity/pull/11763/files#diff-77a7a77f2a2831156ab9f1652dabc5af0858e8ff6739dd3890540ff1d3c5f875

Reply:
OK, so of course - using the `<=shanghai` instead of `=shanghai` will produce a different error for each VM (only available for XXX-compatible), so that's clearly not gonna work in our tests that are run across different EVM versions (which is why Hari removed it in the first place). It is however weird that this doesn't throw the error you mentioned, but triggers a declaration error instead. I'll look into it.

Reply:
I'll leave this as is, and once we make `cancun` the default version in the tests (and compiler), the test should work just fine; this will also have to be set to `=shanghai` due to the error taking into account the fork name, and therefore changing the output depending on the selected EVM version.

Reply:
Yup, I'll ping this to `=shanghai`, but regardless, once we make `cancun` the default, the test will have to be updated since it will yield another error (only available with `cancun` and you're compiling with `xxxxx`).

Reply:
But can you do this also for `basefee` (error 5430)?